event/datadeal/data_verification_event.go

```diff
@@ -1 +1,232 @@
+
+	txBytes, err := generateTxBytes(msgVoteDataVerification, d.reactor.OracleAcc().GetPrivKey(), d.reactor.Config(), txBuilder)
+	if err != nil {
+		return fmt.Errorf("generate tx failed. dealID(%d). dataHash(%s): %w", dealID, dataHash, err)
+	}
+
+	if err := broadcastTx(d.reactor.GRPCClient(), txBytes); err != nil {
+		return fmt.Errorf("broadcast transaction failed. dealID(%d). dataHash(%s): %w", dealID, dataHash, err)
+	}
+
+	return nil
+}
+
+func (d DataVerificationEvent) decryptData(decryptedSharedKey, nonce, encryptedDataBz []byte) ([]byte, error) {
+	decryptedData, err := crypto.DecryptWithAES256(decryptedSharedKey, nonce, encryptedDataBz)
+	if err != nil {
+		return nil, err
+	}
+	return decryptedData, nil
+}
+
+func (d DataVerificationEvent) compareDataHash(dataSale *types.DataSale, decryptedData []byte) bool {
+	decryptedDataHash := sha256.Sum256(decryptedData)
+	decryptedDataHashStr := hex.EncodeToString(decryptedDataHash[:])
+
+	return decryptedDataHashStr == dataSale.DataHash
+}
+
+func (d DataVerificationEvent) convertSellerData(deal *types.Deal, dataSale *types.DataSale) ([]byte, error) {
+	encryptedDataBz, err := d.reactor.Ipfs().Get(dataSale.VerifiableCid)
+	if err != nil {
+		log.Infof("failed to get data from IPFS: %v", err)
+		return nil, nil
+	}
+
+	oraclePrivKey := d.reactor.OraclePrivKey()
+
+	sellerAcc, err := d.reactor.QueryClient().GetAccount(dataSale.SellerAddress)
+	if err != nil {
+		log.Infof("failed to get account from panacea: %v", err)
+		return nil, nil
+	}
+
+	sellerPubKeyBytes := sellerAcc.GetPubKey().Bytes()
+
+	sellerPubKey, err := btcec.ParsePubKey(sellerPubKeyBytes, btcec.S256())
+	if err != nil {
+		log.Infof("failed to parsing sellerPubKey: %v", err)
+		return nil, nil
+	}
+
+	decryptSharedKey := crypto.DeriveSharedKey(oraclePrivKey, sellerPubKey, crypto.KDFSHA256)
+
+	decryptedData, err := d.decryptData(decryptSharedKey, deal.Nonce, encryptedDataBz)
+	if err != nil {
+		return nil, err
+	}
+
+	return decryptedData, nil
+}
+
+func (d DataVerificationEvent) verifyAndGetVoteOption(dealID uint64, dataHash string) (oracletypes.VoteOption, error) {
+	deal, err := d.reactor.QueryClient().GetDeal(dealID)
+	if err != nil {
+		log.Infof("failed to find deal (%d)", dealID)
+		return oracletypes.VOTE_OPTION_NO, nil
+	}
+
+	dataSale, err := d.reactor.QueryClient().GetDataSale(dataHash, dealID)
+	if err != nil {
+		log.Infof("failed to find dataSale (%s)", dataHash)
+		return oracletypes.VOTE_OPTION_UNSPECIFIED, err
+	}
+
+	if dataSale.Status != types.DATA_SALE_STATUS_VERIFICATION_VOTING_PERIOD {
+		return oracletypes.VOTE_OPTION_UNSPECIFIED, errors.New("dataSale's status is not DATA_SALE_STATUS_VERIFICATION_VOTING_PERIOD")
+	}
+
+	decryptedData, err := d.convertSellerData(deal, dataSale)
+	if err != nil {
+		log.Infof("failed to decrypt seller data, error (%v)", err)
+		return oracletypes.VOTE_OPTION_NO, err
+	}
+
+	if !d.compareDataHash(dataSale, decryptedData) {
```

Comment:
if oracle failed get seller's data, this part occur error and it will be vote `NO`. Did I understand correctly?

Reply:
Did you mean the `compareDataHash` method? or the line 158?

Reply:
And based function `convertSellerData`

Reply:
detail lines: https://github.com/medibloc/panacea-doracle/pull/89/files#diff-77b26936f5a1a147ce3878a5f0945af04ae92defbf19d41f4ef201713de9e9e9R109

Reply:
The `compareDataHash` occur errors when after get seller's data and `convertSellerData`.